src/python/grpcio/grpc/_interceptor.py

```diff
@@ -142,23 +194,37 @@ def __init__(self, thunk, method, interceptor):
         self._interceptor = interceptor
 
     def __call__(self, request, timeout=None, metadata=None, credentials=None):
-        call_future = self.future(
+        response, ignored_call = self.with_call(
```

Comment:
Do not use one part of the public API of a system (here the "system" is the object instance) from within the implementation of another part of the same system.

Factor out a shared helper method?

Reply:
Done.